File: 11-container-with-most-water/11-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let done = false;
    let left = 0;
    let right = height.length - 1;
    let area = (right - left) * Math.min(height[left], height[right]); 
    if (height.length == 2) return area; //Base case: arr length is 2
    while (!done) {
        if (height[left] < height[right]) left++;
        else right--;
        if (left == right) return area;
        const tempArea = (right-left) * Math.min(height[left], height[right]); //Recalculate area
        if (tempArea > area) area = tempArea; //If new area >, replace
    }    
};

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int ptr1 = 0;
        int ptr2 = height.length-1;
        int max = 0;
        if (height.length == 2) {
            return Math.min(height[ptr1], height[ptr2]);
        }
        while (ptr2 >= ptr1) {
            if (Math.min(height[ptr1], height[ptr2]) * (ptr2-ptr1) > max) {
                max = Math.min(height[ptr1], height[ptr2]) * (ptr2-ptr1);
            }
            if (height[ptr1] <= height[ptr2]) {
                ptr1++;
            }
            else if (height[ptr1] > height[ptr2]) {
                ptr2--;
            }
        }
        return max;
    }
}